lua/mini/doc.lua

```diff
@@ -971,7 +971,17 @@ H.alias_replace = function(s)
       -- two values which might conflict if outputs are used as arguments.
       local name_escaped = vim.pesc(alias_name)
       local desc_escaped = vim.pesc(alias_desc)
-      s[i] = s[i]:gsub(name_escaped, desc_escaped)
+
+      if string.find(desc_escaped, "|", 1, true) then
+        desc_escaped = "(" .. desc_escaped .. ")"
+      end
+
+      local placeholder = "qqqq" .. name_escaped
+      local section_text = s[i]
+      section_text = section_text:gsub("([_%.]+)%f[%a]" .. name_escaped .. "%f[%A]", "%1" .. placeholder)
```

Comment:
Can't alias names contain digits or underscore? They might not start with digits, but I didn't see any restriction on starting with underscore or ending with underscore/digit. So at least frontier pattern needs to adjust for that. Or even better is to rework the whole replacement to not use them at all. Which might be possible since alias name is used as an explicit Lua pattern and there is no `*` / `-` / `+` quantifiers for which frontier patterns are needed.

Reply:
I think what you've said is a good suggestion but I don't have a lot time to translate your suggestion into working code.

Reply:
No worries. I'll take a look later.